pkg/api/vlabs/validate.go

```diff
@@ -371,6 +371,15 @@ func (a *Properties) ValidateOrchestratorProfile(isUpdate bool) error {
 						}
 					}
 				}
+
+				if o.KubernetesConfig.EtcdStorageLimitGB != 0 {
+					if o.KubernetesConfig.EtcdStorageLimitGB > 8 {
```

Comment:
While I somewhat agree here, I would not put a limit.  etcd recommends not going about 8GB but if someone really needs to play with this value, they should at least look at the limit etcd has.

You could output a warning about that (if you wish) but I would not make it a failure and thus block setting it to, say, 10GB.

NOTE:  I would hope that we never need to go above, say, 4GB as that is rather huge already but the reality is that we don't know where or when this limit may be hit under dynamic cluster conditions.

Reply:
PS - I would not block this PR due to this as it is potentially important to us to get a larger (say 4GB) etcd to test our busy cluster environment.  Just thought it would be nice to not artificially limit the value at this level.

Reply:
Understood, so we want to wrap this behavior:

> 8GB is a suggested maximum size for normal environments and etcd warns at startup if the configured value exceeds it.

Sounds like there are known de-optimizations to the operationalization of the key/val store above 8GB, but there isn't really a hard limit?

Reply:
Done